test/default/vrf.c

```diff
@@ -0,0 +1,135 @@
+
+#define TEST_NAME "vrf"
+#include "cmptest.h"
+
+typedef struct TestData_ {
+    const char seed[2 * 32 + 1];
+    const char pk[2 * 32 + 1];
+    const char proof[2 * 80 + 1];
+    const char output[2 * 64 + 1];
+} TestData;
+/*
+ * Test data taken from https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-vrf-09#appendix-A.4
+ */
+static const TestData test_data[] = {
+        {
+            "9d61b19deffd5a60ba844af492ec2cc44449c5697b326919703bac031cae7f60",
+            "d75a980182b10ab7d54bfed3c964073a0ee172f3daa62325af021a68f707511a",
+            "7d9c633ffeee27349264cf5c667579fc583b4bda63ab71d001f89c10003ab46f14adf9a3cd8b8412d9038531e865c341cafa73589b023d14311c331a9ad15ff2fb37831e00f0acaa6d73bc9997b06501",
+            "9d574bf9b8302ec0fc1e21c3ec5368269527b87b462ce36dab2d14ccf80c53cccf6758f058c5b1c856b116388152bbe509ee3b9ecfe63d93c3b4346c1fbc6c54",
+        },
+        {
+            "4ccd089b28ff96da9db6c346ec114e0f5b8a319f35aba624da8cf6ed4fb8a6fb",
+            "3d4017c3e843895a92b70aa74d1b7ebc9c982ccf2ec4968cc0cd55f12af4660c",
+            "47b327393ff2dd81336f8a2ef10339112401253b3c714eeda879f12c509072ef055b48372bb82efbdce8e10c8cb9a2f9d60e93908f93df1623ad78a86a028d6bc064dbfc75a6a57379ef855dc6733801",
+            "38561d6b77b71d30eb97a062168ae12b667ce5c28caccdf76bc88e093e4635987cd96814ce55b4689b3dd2947f80e59aac7b7675f8083865b46c89b2ce9cc735",
+        },
+        {
+            "c5aa8df43f9f837bedb7442f31dcb7b166d38535076f094b85ce3a2e0b4458f7",
+            "fc51cd8e6218a1a38da47ed00230f0580816ed13ba3303ac5deb911548908025",
+            "926e895d308f5e328e7aa159c06eddbe56d06846abf5d98c2512235eaa57fdce35b46edfc655bc828d44ad09d1150f31374e7ef73027e14760d42e77341fe05467bb286cc2c9d7fde29120a0b2320d04",
+            "121b7f9b9aaaa29099fc04a94ba52784d44eac976dd1a3cca458733be5cd090a7b5fbd148444f17f8daf1fb55cb04b1ae85a626e30a54b4b0f8abf4a43314a58",
+        }
+};
+
+static const unsigned char messages[3][2] = {{0x00}, {0x72}, {0xaf, 0x82}};
+
+static inline void printhex(const char *label, const unsigned char *c, size_t len){
+    size_t i;
+    printf("%s", label);
+    for (i = 0; i < len; i++){
+        printf("%02x", c[i]);
+    }
+    printf("\n");
+}
+
+int main(void)
+{
+    unsigned char *seed, *expected_pk, *expected_proof, *expected_output;
+    seed            = (unsigned char *) sodium_malloc(crypto_vrf_ietfdraft12_SEEDBYTES);
+    expected_pk     = (unsigned char *) sodium_malloc(crypto_vrf_ietfdraft12_PUBLICKEYBYTES);
+    expected_proof  = (unsigned char *) sodium_malloc(crypto_vrf_ietfdraft12_BYTES);
+    expected_output = (unsigned char *) sodium_malloc(crypto_vrf_ietfdraft12_OUTPUTBYTES);
+
+    unsigned char sk[64];
```

Comment:
libsodium uses C89, since some supported compilers still don't support C99 yet.

Would you mind moving all the declarations to the top of the block?